redis/src/main/java/org/hibernate/ogm/datastore/redis/dialect/model/impl/DocumentBasedAssociation.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.redis.dialect.model.impl;
+
+import java.util.List;
+
+import org.hibernate.ogm.datastore.redis.dialect.value.Association;
+import org.hibernate.ogm.datastore.redis.dialect.value.StructuredValue;
+
+
+/**
+ * A {@link RedisAssociation} backed by an {@link Association}.
+ *
+ * @author Gunnar Morling
+ */
+class DocumentBasedAssociation extends RedisAssociation {
```

Comment:
Where should I put the classes? In `org.hibernate.ogm.datastore.document.association.spi` and `org.hibernate.ogm.datastore.association.spi`?

What about the name? Would go for `AssociationRepresentation`, `EmbeddedAssociationRepresentation` and `DocumentBasedAssociationRepresentation`, each typed with `T` for `T getOwningDocument()`.

But to be honest, the whole construction of the `RedisAssociation` and `CouchDBAssociation` is sort of strange. It's just a bridge between `AssociationRows` and the representation. We should discuss that topic, perhaps we can build something just on top of `AssociationRows`.